nearcore/src/state_sync.rs

```diff
@@ -487,147 +494,28 @@ fn start_dumping(
     let state_header = chain.get_state_response_header(shard_id, sync_hash)?;
     let num_parts = get_num_state_parts(state_header.state_root_node().memory_usage);
     if shard_tracker.care_about_shard(account_id.as_ref(), sync_prev_prev_hash, shard_id, true) {
-        tracing::info!(target: "state_sync_dump", shard_id, ?epoch_id, %sync_prev_hash, %sync_hash, "Initialize dumping state of Epoch");
+        tracing::info!(target: "state_sync_dump", shard_id, ?epoch_id, epoch_height, %sync_prev_hash, %sync_hash, "Dumping state of Epoch");
         // Note that first the state of the state machines gets changes to
         // `InProgress` and it starts dumping state after a short interval.
         set_metrics(&shard_id, Some(0), Some(num_parts), Some(epoch_height));
         Ok(Some(StateSyncDumpProgress::InProgress { epoch_id, epoch_height, sync_hash }))
     } else {
-        tracing::info!(target: "state_sync_dump", shard_id, ?epoch_id, %sync_hash, "Shard is not tracked, skip the epoch");
-        Ok(Some(StateSyncDumpProgress::AllDumped { epoch_id, epoch_height, num_parts: Some(0) }))
+        tracing::info!(target: "state_sync_dump", shard_id, ?epoch_id, epoch_height, %sync_hash, "Shard is not tracked, skip the epoch");
+        Ok(Some(StateSyncDumpProgress::AllDumped { epoch_id, epoch_height }))
     }
 }
 
-/// Checks what is the latest complete epoch.
-/// `epoch_id` represents the last fully dumped epoch.
-fn check_new_epoch(
-    epoch_id: Option<EpochId>,
-    epoch_height: Option<EpochHeight>,
-    num_parts: Option<u64>,
+/// return epoch_id and sync_hash of the latest complete epoch available locally.
+fn check_latest_epoch(
```

Comment:
Should this be renamed to `get_latest_epoch()` or similar?

Reply:
sounds good